src/app/components/home/card-list/card-list.component.ts

```diff
@@ -0,0 +1,182 @@
+import { SearchBarEventArgs } from '../../../models/nav-bar/search-bar-event-args';
+import { NgxUiLoaderService } from 'ngx-ui-loader';
+import { PokemonsPage } from '../../../models/shared/pokemons-page';
+import { Pokemon } from '../../../models/shared/pokemon';
+import { FetchedPokemonsEntityService } from '../../../services/fetched-pokemons/fetched-pokemons-entity.service';
+import { Observable, Subscription } from 'rxjs';
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { PokemonEntityService } from '../../../services/pokemons-page/pokemon-entity.service';
+import { map, tap } from 'rxjs/operators';
+import { defaultDialogConfig } from '../../../shared/default-dialog-config';
+import { MatDialog } from '@angular/material/dialog';
+import { SingleCardOverviewComponent } from '../single-card-overview/single-card-overview.component';
+import { MultipleCardOverviewComponent } from '../multiple-card-overview/multiple-card-overview.component';
+import { FavoriteEntityService } from 'src/app/services/favorite-pokemons/favorite-entity.service';
+
+@Component({
+  selector: 'app-card-list',
+  templateUrl: './card-list.component.html',
+  styleUrls: ['./card-list.component.scss'],
+})
+export class CardListComponent implements OnInit, OnDestroy {
+  pokemonsPage$: Observable<PokemonsPage[]>;
+  isComparing: boolean = false;
+  pokemonBeforeComparing: Pokemon;
+  pokemonApiOffset: string = '20';
+  queryParams: string = '';
+
+  favoriteSubscription: Subscription;
+  fetchedPokemonsSubscription: Subscription;
+  dialogSubscription: Subscription;
+
+  constructor(
+    private pokemonsEntityService: PokemonEntityService,
+    private dialog: MatDialog,
+    private fetchedPokemonsEntityService: FetchedPokemonsEntityService,
+    private favoriteEntityService: FavoriteEntityService,
+    private loader: NgxUiLoaderService
+  ) {}
+
+  ngOnInit(): void {
+    this.loadStoredPokemons();
+
+    this.pokemonsEntityService.entities$.pipe(
+      tap((item) => {
+        this.pokemonApiOffset = item.length + '';
+      })
+    );
+  }
+
+  ngOnDestroy() {
+    this.tryToUnsuscribe(this.favoriteSubscription);
+    this.tryToUnsuscribe(this.fetchedPokemonsSubscription);
+    this.tryToUnsuscribe(this.dialogSubscription);
```

Comment:
if you have multiple subscriptions in the same file, use the take until pipe instead